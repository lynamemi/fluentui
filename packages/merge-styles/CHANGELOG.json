{
  "name": "@uifabric/merge-styles",
  "entries": [
    {
      "version": "5.4.0",
      "tag": "@uifabric/merge-styles_v5.4.0",
      "date": "Fri, 13 Oct 2017 04:00:17 GMT",
      "comments": {
        "patch": [
          {
            "author": "David Zearing <dzearing@microsoft.com>",
            "commit": "2b318265a85acf7d858bcd5b161641e6c2d7c07a",
            "comment": "Adding data-merge-styles attribute to the style element so that it can be uniquely identified."
          }
        ],
        "minor": [
          {
            "author": "David Zearing <dzearing@microsoft.com>",
            "commit": "2b318265a85acf7d858bcd5b161641e6c2d7c07a",
            "comment": "Fixes to support media queries."
          }
        ]
      }
    },
    {
      "version": "5.3.3",
      "tag": "@uifabric/merge-styles_v5.3.3",
      "date": "Fri, 13 Oct 2017 01:36:02 GMT",
      "comments": {
        "patch": [
          {
            "author": "David Zearing <dzearing@microsoft.com>",
            "commit": "1c74d9f0c7d7af5edada08e69e0d71968c90f13f",
            "comment": "Adding data-merge-styles attribute to the style element so that it can be uniquely identified."
          }
        ]
      }
    },
    {
      "version": "5.3.2",
      "tag": "@uifabric/merge-styles_v5.3.2",
      "date": "Mon, 09 Oct 2017 10:08:09 GMT",
      "comments": {
        "patch": [
          {
            "author": "Christian Gonzalez <christianjordangonzalez@gmail.com>",
            "commit": "71ff33573d03e3cdefc2ba2c5410711be6e2ee09",
            "comment": "Add test for autoexpand"
          }
        ]
      }
    },
    {
      "version": "5.3.1",
      "tag": "@uifabric/merge-styles_v5.3.1",
      "date": "Fri, 06 Oct 2017 10:18:41 GMT",
      "comments": {
        "patch": [
          {
            "author": "Jon Schectman <joschect@microsoft.com>",
            "commit": "8e1dcc66cda8334653a8b5ccbde02bf6275e0b23",
            "comment": "TSConfig: update to use preserveConstEnums so that certain builds s ystems don't break when importing const enums"
          }
        ]
      }
    },
    {
      "version": "5.3.0",
      "tag": "@uifabric/merge-styles_v5.3.0",
      "date": "Thu, 05 Oct 2017 10:17:42 GMT",
      "comments": {
        "minor": [
          {
            "author": "David Zearing <dzearing@microsoft.com>",
            "commit": "f95fa74185b6faaf278fdc0a78e72635bcc3ec48",
            "comment": "Adding selector support to target child class names that were generated in the same mergeStyleSets set."
          }
        ]
      }
    },
    {
      "version": "5.2.0",
      "tag": "@uifabric/merge-styles_v5.2.0",
      "date": "Sat, 30 Sep 2017 01:26:37 GMT",
      "comments": {
        "minor": [
          {
            "author": "David Zearing <dzearing@microsoft.com>",
            "commit": "4ef41283e7667631964dd6cf973534fab4cbb1fa",
            "comment": "Found that adding `false` to the list of accepted typings in `mergeStyleSets` was breaking type safety. Removed it."
          }
        ]
      }
    },
    {
      "version": "5.1.0",
      "tag": "@uifabric/merge-styles_v5.1.0",
      "date": "Fri, 29 Sep 2017 10:20:24 GMT",
      "comments": {
        "minor": [
          {
            "author": "David Zearing <dzearing@microsoft.com>",
            "commit": "88b06cfda2262661375f90f3bde2d6a16af749a9",
            "comment": "The `mergeStyleSets` method's type safety was not correct. Now with significantly better type safety."
          }
        ]
      }
    }
  ]
}