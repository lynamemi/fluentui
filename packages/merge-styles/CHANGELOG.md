# Change Log - @uifabric/merge-styles

This log was last generated on Fri, 13 Oct 2017 04:00:17 GMT and should not be manually modified.

## 5.4.0
Fri, 13 Oct 2017 04:00:17 GMT

### Minor changes

- Fixes to support media queries.

### Patches

- Adding data-merge-styles attribute to the style element so that it can be uniquely identified.

## 5.3.3
Fri, 13 Oct 2017 01:36:02 GMT

### Patches

- Adding data-merge-styles attribute to the style element so that it can be uniquely identified.

## 5.3.2
Mon, 09 Oct 2017 10:08:09 GMT

### Patches

- Add test for autoexpand

## 5.3.1
Fri, 06 Oct 2017 10:18:41 GMT

### Patches

- TSConfig: update to use preserveConstEnums so that certain builds s ystems don't break when importing const enums

## 5.3.0
Thu, 05 Oct 2017 10:17:42 GMT

### Minor changes

- Adding selector support to target child class names that were generated in the same mergeStyleSets set.

## 5.2.0
Sat, 30 Sep 2017 01:26:37 GMT

### Minor changes

- Found that adding `false` to the list of accepted typings in `mergeStyleSets` was breaking type safety. Removed it.

## 5.1.0
Fri, 29 Sep 2017 10:20:24 GMT

### Minor changes

- The `mergeStyleSets` method's type safety was not correct. Now with significantly better type safety.

