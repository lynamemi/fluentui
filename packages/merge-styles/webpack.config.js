const path = require('path');
const resources = require('../../scripts/tasks/webpack-resources');
const webpack = resources.webpack;

const BUNDLE_NAME = 'merge-styles';

module.exports = resources.createConfig(BUNDLE_NAME, true, {
  entry: {
    [BUNDLE_NAME]: './lib/index.js'
  },

  output: {
    libraryTarget: 'var',
    library: 'MergeStyles'
  }
});
