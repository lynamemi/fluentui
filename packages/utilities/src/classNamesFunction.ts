import { mergeStyleSets, IStyle } from '@uifabric/merge-styles';
import { IClassNames } from './IClassNames';
import { IStyleFunction } from './IStyleFunction';
import { IStyleFunctionOrObject } from './styled';

/**
 * Creates a getClassNames function which calls getStyles given the props, and injects them
 * into mergeStyleSets.
 */
export function classNamesFunction<TStyleProps extends {}, TStyles extends { [P in keyof TStyles]: IStyle }>(): (
  getStyles?: IStyleFunctionOrObject<TStyleProps, TStyles>,
  styleProps?: TStyleProps
) => IClassNames<TStyles> {
  // TODO: memoize.
  return (getStyles?: IStyleFunctionOrObject<TStyleProps, TStyles>, styleProps?: TStyleProps): IClassNames<TStyles> =>
    mergeStyleSets(getStyles && (typeof getStyles === 'function' ? getStyles(styleProps!) : getStyles));
}
