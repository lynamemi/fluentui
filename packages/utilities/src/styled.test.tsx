import * as React from 'react';
import { styled, IStyleFunctionOrObject } from './styled';
import * as renderer from 'react-test-renderer';
import { Customizer } from './Customizer';
import { IStyle, Stylesheet, InjectionMode } from '@uifabric/merge-styles';
import { classNamesFunction } from './classNamesFunction';

interface ITestStyles {
  root: IStyle;
}
interface ITestProps {
  cool?: boolean;
  styles?: IStyleFunctionOrObject<{}, ITestStyles>;
}

let _lastProps: ITestProps | undefined;

const getClassNames = classNamesFunction<{}, ITestStyles>();

class TestBase extends React.Component<ITestProps> {
  public constructor(props: ITestProps) {
    super(props);
    _lastProps = props;
  }
  public render(): JSX.Element {
    const classNames = getClassNames(this.props.styles, this.props);

    return <div className={classNames.root} />;
  }
}

const TestStyles = (props: ITestProps): ITestStyles => ({
  root: {
    background: props.cool ? 'blue' : 'red'
  }
});

const Test = styled<ITestProps, {}, ITestStyles>(TestBase, TestStyles, undefined, { scope: 'Test' });

describe('styled', () => {
  beforeEach(() => {
    _lastProps = undefined;
    Stylesheet.getInstance().setConfig({
      injectionMode: InjectionMode.none
    });
    Stylesheet.getInstance().reset();
  });

  it('renders base styles (background red)', () => {
    const component = renderer.create(<Test />);

    // Test that defaults are the base styles (red).
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('allows user overrides (background green)', () => {
    const component = renderer.create(<Test styles={{ root: { background: 'green' } }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('allows for contextual overrides (background yellow)', () => {
    const component = renderer.create(
      <Customizer
        scopedSettings={{
          Test: {
            styles: {
              root: {
                background: 'yellow'
              }
            }
          }
        }}
      >
        <Test />
      </Customizer>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('allows for contextual overrides mixed under user overrides (background yellow, color red)', () => {
    const component = renderer.create(
      <Customizer
        scopedSettings={{
          Test: {
            styles: {
              root: {
                background: 'yellow'
              }
            }
          }
        }}
      >
        <Test styles={{ root: { color: 'red' } }} />
      </Customizer>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('can process style props (background blue)', () => {
    const component = renderer.create(<Test cool />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
