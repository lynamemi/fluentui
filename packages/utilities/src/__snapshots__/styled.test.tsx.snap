// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styled allows for contextual overrides (background yellow) 1`] = `
<div
  className=

      {
        background: yellow;
      }
/>
`;

exports[`styled allows for contextual overrides mixed under user overrides (background yellow, color red) 1`] = `
<div
  className=

      {
        background: yellow;
        color: red;
      }
/>
`;

exports[`styled allows user overrides (background green) 1`] = `
<div
  className=

      {
        background: green;
      }
/>
`;

exports[`styled can process style props (background blue) 1`] = `
<div
  className=

      {
        background: blue;
      }
/>
`;

exports[`styled renders base styles (background red) 1`] = `
<div
  className=

      {
        background: red;
      }
/>
`;
